ts/features/messages/hooks/useAttachmentDownload.tsx

```diff
@@ -37,37 +75,28 @@ export const useAttachmentDownload = (
       void mixpanelTrack("PN_ATTACHMENT_DOWNLOADFAILURE");
       showToast(i18n.t("messageDetails.attachments.failing.details"));
     } else if (download) {
-      const path = download.path;
-      const attachment = download.attachment;
+      const { path, attachment } = download;
+
       if (attachment.contentType === ContentTypeValues.applicationPdf) {
         openPreview(attachment);
       } else {
-        if (isIos) {
-          ReactNativeBlobUtil.ios.presentOptionsMenu(path);
-        } else {
-          try {
-            const downloadFilePath =
-              await ReactNativeBlobUtil.MediaCollection.copyToMediaStore(
-                {
-                  name: attachment.displayName,
-                  parentFolder: "",
-                  mimeType: attachment.contentType
-                },
-                "Download",
-                path
-              );
-
-            await ReactNativeBlobUtil.android.addCompleteDownload({
-              mime: attachment.contentType,
-              title: attachment.displayName,
-              showNotification: true,
-              description: attachment.displayName,
-              path: downloadFilePath
-            });
-          } catch (e) {
-            showToast(i18n.t("messageDetails.attachments.failing.details"));
-          }
-        }
+        await pipe(
+          isAndroid,
+          O.fromPredicate(identity),
```

Comment:
Can we replace `O.fromPredicate` and `TE.fromOption` with just `TE.fromPredicate((_: boolean) => _, () => undefined)`?